### PR TITLE
[DOCS] Add New Topic for Identifying Failed Table Rows

### PR DESCRIPTION
**Pull request by kwcanuck**
In [DOC-541](https://greatexpectations.atlassian.net/browse/DOC-541), @Rachel-Reverie requested the addition of a new topic that documents the process of identifying failed rows in Expectations. This PR adds the new topic.

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] _Appropriate_ tests and docs have been updated

[DOC-541]: https://greatexpectations.atlassian.net/browse/DOC-541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8554&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=8554&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+<TabItem value="spark">
+
+The following is an example of the information returned after running the Checkpoint. For Spark, a filter condition on the DataFrame is included.
+
+</TabItem>
+<TabItem value="sqlalchemy">
+
+The following is an example of the information returned after running the Checkpoint. For SQLAlchemy, the utput includes a query that can be used directly against the database backend.
+
+</TabItem>
```
Comment (austiezr) [inline docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md]: These are missing example code.

**Review by austiezr — CHANGES_REQUESTED** [review]
No problems with content -- we are missing example code for two tabs at the end, not certain how necessary it is to be explicit with all possible backends for output.

File: docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+<TabItem value="spark">
+
+The following is an example of the information returned after running the Checkpoint. For Spark, a filter condition on the DataFrame is included.
+
+</TabItem>
+<TabItem value="sqlalchemy">
+
+The following is an example of the information returned after running the Checkpoint. For SQLAlchemy, the utput includes a query that can be used directly against the database backend.
```
Comment (kwcanuck) [inline docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md]:
```suggestion
The following is an example of the information returned after running the Checkpoint. For SQLAlchemy, the output includes a query that can be used directly against the database backend.
```

File: docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+### Import the Checkpoint 
+
+Run the following code to import the example Checkpoint:
+
+```python name="tests/integration/docusaurus/expectations/advanced/failed_rows_pandas.py get checkpoint"
+```
+
+### Set the `unexpected_index_column_names` parameter
+
+The failed rows are defined as values in the `unexpected_index_column_names` parameter.  In the following example, you are setting the parameter to `event_id` to return the `event_ids` of the rows that fail the Expectation. However, `unexpected_index_column_names` can also return a list of columns that are representative of the failed rows. The Checkpoint also returns the `unexpected_index_query`, which you can use to retrieve the full set of failed results.
```
Comment (Shinnnyshinshin) [inline docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md:66]: Thank you for this @kwcanuck :) love it

File: docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+### Run Checkpoint using `result_format`
+
+Run the following code to apply the updated `result_format` to every Expectation in your Suite and pass it directly to the `run()` method: 
+
+```python name="tests/integration/docusaurus/expectations/advanced/failed_rows_pandas.py run checkpoint"
+```
+
+### Review Validation Results
+
+After you run the Checkpoint, a dictionary for each failed row is returned. Each dictionary contains the row’s identifier and the failed value.
```
Comment (Shinnnyshinshin) [inline docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md:82]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM pending the clarification of the "dropdown" term that you mentioned @kwcanuck

**Comment by kwcanuck** [comment]
@Shinnnyshinshin thanks for the quick review and your infinite patience! The content is now ready for your final review and (hopefully) approval.

File: docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+### Open the Data Docs
+
+Run the following code to open the Data Docs:
+
+```python
+context.open_data_docs()
+```
+
+The following image shows the `unexpected_index_list` values are displayed as part of the validation output. Click **To retrieve all unexpected values** to view the SQL query that you can use to retrieve all unexpected values.
```
Comment (Shinnnyshinshin) [inline docs/docusaurus/docs/guides/expectations/advanced/identify_failed_rows_expectations.md:301]: Looks great :)

**Review by austiezr — APPROVED** [review]
LGTM!

**Review by Rachel-Reverie — APPROVED** [review]
LGTM

**Pull request merged**